### DISCUSSION TITLE
I installed Skywalking on "Kubernetes Version: v1.27.10 +rke2r1" but my dashboard is all empty, what am i missing?

### DISCUSSION THREAD

**Discussion by 0nurulusoy** [apache/skywalking#12314] [Q&A] [answered]
I installed Skywalking on Kubernetes (Version: v1.27.10 +rke2r1) using the following commands:

```
export SKYWALKING_RELEASE_VERSION=4.6.0
export SKYWALKING_RELEASE_NAME=skywalking
export SKYWALKING_RELEASE_NAMESPACE=skywalking

helm install "${SKYWALKING_RELEASE_NAME}" \
  oci://registry-1.docker.io/apache/skywalking-helm \
  --version "${SKYWALKING_RELEASE_VERSION}" \
  -n "${SKYWALKING_RELEASE_NAMESPACE}" \
  --set oap.image.tag=10.0.0 \
  --set oap.storageType=elasticsearch \
  --set ui.image.tag=10.0.0
```

It works fine and all pods are healthy, however my Skywalking dashboard is empty, it didn't discover nodes or services. What could be missing?

<img width="1915" alt="image" src="https://github.com/apache/skywalking/assets/55711122/81ebb92e-da3b-455d-b14a-1fd98010cd72">

**Comment by wu-sheng** [comment, marked as answer]
Do you follow the docs to install relative components? OAP itself would not collect metrics from k8s.

https://skywalking.apache.org/docs/main/latest/en/setup/backend/backend-k8s-monitoring/

**Reply by 0nurulusoy** [reply]
Thanks, I just readed the document, and I need Rover, is it possible to monitor and trace applications without agents inside application? Can it auto discover apps on Kubernetes and trace them without an agent, all the metrics for K8S monitoing are system level which i don't need.

**Reply by wu-sheng** [reply]
Momitoring, yes. Tracing, no.
All tracing relies on in process agent to propagate tracing context.

**Reply by 0nurulusoy** [reply]
Thanks, is there a document for Java, that show how to add agent to Kubernetes Pod? I know very basic config but since agent is a Jar, do i have to add jar and run -javaagent: to every pod's Dockerfile?

**Reply by wu-sheng** [reply]
Your ways work.

Also, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/containerization/#docker
